cmake/PureCMakeBuild.cmake

```diff
@@ -0,0 +1,290 @@
+)
+
+# optionally, opencl
+# OpenCL disabled as its code is not up-to-date with API
+set(USE_OPEN_CL FALSE CACHE BOOL "Set to TRUE to look for OpenCL")
+if (USE_OPEN_CL)
+	find_path(OPENCL_INCLUDE_DIR CL/cl.h
+		/usr/local/include
+		/usr/include
+	)
+	if (WIN32)
+		find_library(OPENCL_LIBRARIES opencl64)
+		if (!OPENCL_LIBRARIES)
+			find_library(OPENCL_LIBRARIES opencl32)
+		endif ()
+	else ()
+		find_library(OPENCL_LIBRARIES OpenCL ENV LD_LIBRARY_PATH)
+	endif ()
+	if (OPENCL_INCLUDE_DIR AND OPENCL_LIBRARIES)
+		add_definitions(-DHAVE_OPENCL)
+		set(EXTRA_LIBS ${OPENCL_LIBRARIES} ${EXTRA_LIBS})
+		include_directories(${OPENCL_INCLUDE_DIR})
+		add_definitions(-DOPENCL_SOURCE_DIR=\"${CMAKE_SOURCE_DIR}/nabo/opencl/\")
+		message("OpenCL enabled and found, enabling CL support")
+	else (OPENCL_INCLUDE_DIR AND OPENCL_LIBRARIES)
+		message("OpenCL enabled but not found, disabling CL support")
+	endif ()
+else ()
+	message("OpenCL disabled, not looking for it")
+endif ()
+
+set(SHARED_LIBS FALSE CACHE BOOL "Set to TRUE to build shared library")
+if (SHARED_LIBS)
+	add_library(${LIB_NAME} SHARED ${NABO_SRC})
+	install(TARGETS ${LIB_NAME} LIBRARY DESTINATION lib)
+else ()
+	add_library(${LIB_NAME} STATIC ${NABO_SRC})
+	if (NOT MSVC)
+		target_compile_options(${LIB_NAME} PRIVATE -fPIC)
+	endif()
+	install(TARGETS ${LIB_NAME} ARCHIVE DESTINATION lib)
+endif ()
+set_target_properties(${LIB_NAME} PROPERTIES VERSION "${PROJECT_VERSION}" SOVERSION 1)
+
+target_include_directories(${LIB_NAME} PUBLIC
+	${EIGEN_INCLUDE_DIR}
+	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  )
+
+# openmp
+set(USE_OPEN_MP TRUE CACHE BOOL "Set to FALSE to not use OpenMP")
+if (USE_OPEN_MP)
+	find_package(OpenMP)
+	if (OPENMP_FOUND)
+		target_compile_options(${LIB_NAME} PRIVATE -fopenmp ${OpenMP_CXX_FLAGS})
+		target_compile_definitions(${LIB_NAME} PRIVATE HAVE_OPENMP)
+		if (CMAKE_COMPILER_IS_GNUCC)
+			target_link_libraries(${LIB_NAME} PUBLIC gomp)
+		endif()
+	endif()
+endif ()
+
+
+# create doc before installing
+set(DOC_INSTALL_TARGET "share/doc/${PROJECT_NAME}/api" CACHE STRING "Target where to install doxygen documentation")
+install(FILES nabo/nabo.h DESTINATION include/nabo)
+install(FILES nabo/third_party/any.hpp DESTINATION include/nabo/third_party)
+install(FILES README.md DESTINATION share/doc/${PROJECT_NAME})
+if (DOXYGEN_FOUND)
+	install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/doc/html DESTINATION ${DOC_INSTALL_TARGET})
+endif (DOXYGEN_FOUND)
+
+enable_testing()
+
+
+option(LIBNABO_BUILD_EXAMPLES "Build libnabo examples" ON)
+if (LIBNABO_BUILD_EXAMPLES)
+  add_subdirectory(examples)
+endif()
+
+option(LIBNABO_BUILD_TESTS "Build libnabo tests" ON)
+if(LIBNABO_BUILD_TESTS)
+  add_subdirectory(tests)
+endif()
+
+option(LIBNABO_BUILD_PYTHON "Build libnabo python" ON)
+if(LIBNABO_BUILD_PYTHON)
+  add_subdirectory(python)
+endif()
+
+# Install catkin package.xml
+install(FILES package.xml DESTINATION share/libnabo)
```

Comment:
I wouldn't do this. Either the package is built with catkin as a ROS package, or it is built as a pure CMake package. But mixing these would require that pure-cmake installed cmake files would need to be catkin-compatible.

Reply:
If you want to retain this, then COMPONENT dev.